### PR TITLE
Bugfix: Update valid hostname value in iosxr_facts

### PR DESCRIPTION
**Pull request by ashwini-mhatre**
##### SUMMARY

Update valid hostname value in iosxr_facts using show running-conf hostname commands

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

iosxr_facts
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: plugins/cliconf/iosxr.py [hunk 1]
```diff
@@ -94,7 +95,7 @@ def get_device_info(self):
                     device_info["network_os_model"] = match.group(1)
                     break
 
-            match = re.search(r"^(.+) uptime", data, re.M)
+            match = re.search(r"hostname\s(\S+)$", hostname, re.M)
```
Comment (NilashishC) [inline plugins/cliconf/iosxr.py:98]: We should have an unit test that validates that this change does not break this method for older versions.

**Comment by ashwini-mhatre** [comment]
recheck

**Review by NilashishC — APPROVED** [review]

File: plugins/module_utils/network/iosxr/argspec/ospf_interfaces/ospf_interfaces.py [hunk 2]
```diff
@@ -76,19 +76,27 @@ def __init__(self, **kwargs):
...
                                 },
                                 "null_auth": {"type": "bool"},
                             },
                         },
                         "authentication_key": {
                             "type": "dict",
+                            "no_log": False,
                             "options": {
-                                "password": {"type": "str"},
+                                "password": {"type": "str", "no_log": False},
```
Comment (NilashishC) [inline plugins/module_utils/network/iosxr/argspec/ospf_interfaces/ospf_interfaces.py:99]: Please hold on to merging the no_log changes till we take a final call.

cc @GomathiselviS

Comment (NilashishC) [reply plugins/module_utils/network/iosxr/argspec/ospf_interfaces/ospf_interfaces.py:99]: Rest LGTM!

Comment (NilashishC) [reply plugins/module_utils/network/iosxr/argspec/ospf_interfaces/ospf_interfaces.py:99]: @ashwini-mhatre As decided, we are masking all keys that may have sensitive information as values. Once that is handled, please feel free to merge the PR.

**Comment by ashwini-mhatre** [comment]
recheck

**Comment by ashwini-mhatre** [comment]
recheck

**Comment by ashwini-mhatre** [comment]
recheck

**Comment by ashwini-mhatre** [comment]
recheck

**Comment by ashwini-mhatre** [comment]
recheck

File: requirements.txt [hunk 3]
```diff
@@ -1,4 +1,3 @@
-lxml
```
Comment (pabelanger) [inline requirements.txt:1]: why was this removed? It would seem that lxml isn't a dependency then?

Comment (ashwini-mhatre) [reply requirements.txt:1]: ncclient will install lxml as dependancy

Comment (ashwini-mhatre) [reply requirements.txt:1]: actually we need lxml for ncclient installation but as ncclient itself will take care of this i think this is not needed and also after removing this also we have lxml installed properly for more clarification please check this https://github.com/ncclient/ncclient/blob/master/requirements.txt

Comment (ashwini-mhatre) [reply requirements.txt:1]: also please check this https://ac974acc6c5551280ed4-b3f9d1886e750b23ce2ba8d42dd6b9f6.ssl.cf1.rackcdn.com/117/e5c1bf02cf428fd36a4c683d670e8a275f32e08b/check/ansible-test-units-iosxr-python35/396ada6/job-output.txt

Comment (ashwini-mhatre) [reply requirements.txt:1]: @pabelanger shall I add gate?

File: plugins/module_utils/network/iosxr/argspec/ospf_interfaces/ospf_interfaces.py [hunk 4]
```diff
@@ -311,13 +319,17 @@ def __init__(self, **kwargs):
...
                             "options": {
                                 "id": {"type": "int", "required": True},
                                 "md5": {
                                     "type": "dict",
                                     "required": True,
                                     "options": {
-                                        "password": {"type": "str"},
+                                        "password": {
+                                            "type": "str",
+                                            "no_log": False,
```
Comment (NilashishC) [inline plugins/module_utils/network/iosxr/argspec/ospf_interfaces/ospf_interfaces.py:331]: Should this be True?

Comment (ashwini-mhatre) [reply plugins/module_utils/network/iosxr/argspec/ospf_interfaces/ospf_interfaces.py:331]: will address this in https://github.com/ansible-collections/cisco.iosxr/pull/123

**Pull request merged**